lib/src/options.dart

```diff
@@ -436,3 +447,51 @@ class CommandLineParser {
     return i;
   }
 }
+
+/// Parses options from a YAML-format options file.
+///
+/// The options file format is intentionally very open-ended, giving clients
+/// utmost flexibility in defining their own options.  The only hardfast
+/// expectation is that options files will contain a mapping from Strings
+/// (identifying 'scopes') to associated options.  For example, the given
+/// content
+///
+///     linter:
+///       rules:
+///         camel_case_types: true
+///     compiler:
+///       resolver:
+///         useMultiPackage: true
+///       packagePaths:
+///         - /foo/bar/pkg
+///         - /bar/baz/pkg
+///
+/// defines two scopes, `linter` and `compiler`.  Parsing would result in a
+/// map, mapping the `linter` and `compiler` scope identifiers to their
+/// respective parsed option node contents. Extracting values is a simple
+/// matter of inspecting the parsed nodes.  For example, testing whether the
+/// compiler's resolver is set to use the `useMultiPackage` option might look
+/// something like this (eliding error-checking):
+///
+///     bool useMultiPackage =
+///         options['compiler']['resolver']['useMultiPackage'];
+class OptionsFileParser {
+  /// Parse the given [yaml] source into a mapping of option scopes to
+  /// associated option content [YamlNode]s.
+  Map<String, YamlNode> parse(String yaml) {
+    var options = <String, YamlNode>{};
+    var doc = loadYaml(yaml);
+    if (doc is! YamlMap) {
+      throw 'Bad options file format (expected map, got ${doc.runtimeType})';
```

Comment:
Personally, I consider it to be bad form to throw something other than an exception, and it violates the contract we set up with OptionsProcessor. (Both here and below.)